tsl/src/data_node.c

```diff
@@ -2056,14 +2056,28 @@ Datum
 data_node_ping(PG_FUNCTION_ARGS)
 {
 	const char *node_name = PG_ARGISNULL(0) ? NULL : PG_GETARG_CSTRING(0);
+	Interval *timeout = PG_ARGISNULL(1) ? NULL : PG_GETARG_INTERVAL_P(1);
+	TimestampTz endtime = TS_NO_TIMEOUT;
+
 	/* Allow anyone to ping a data node. Otherwise the
 	 * timescaledb_information.data_node view won't work for those users. */
 	ForeignServer *server = data_node_get_foreign_server(node_name, ACL_NO_CHECK, false, false);
 	bool success;
 
 	Assert(NULL != server);
 
-	success = remote_connection_ping(server->servername);
+	if (timeout)
+	{
+		if (timeout->day || timeout->month)
+			ereport(ERROR,
+					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+					 errmsg("ping interval timeout is too large")));
```

Comment:
Why should we check this and make the decision a timeout is too big? What's the use case?

Reply:
I think a more reasonable sanity check is to ensure that ping timeout is always smaller than statement_timeout. Otherwise, the ping function will always time out on statement_timeout before reaching its own timeout.

Reply:
Shouldn't this be expected? Ok, computing month and day now.